src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,556 @@
+
+    """
+    sample_hash = hashlib.sha256()
+
+    for k, v in sorted(sample.items()):
+        sample_hash.update(k.encode("utf8"))
+
+        if isinstance(v, dict):
+            sample_hash.update(compute_identity(size, **v).encode("utf8"))
+        else:
+            sample_hash.update(str(v).encode("utf8"))
+
+    return sample_hash.hexdigest()[:size]
+
+
+def save_task_network(
+    checkpoint_file: Union[str, Path], benchmark: str, network: nn.Module, h: np.ndarray
+) -> None:
+    checkpoint_file = Path(checkpoint_file)
+    state = dict(
+        benchmark=benchmark,
+        network=network.state_dict(),
+        size=list(network.parameters())[0].size()[1],
+        h=h.tolist(),
+    )
+
+    tmp_file = checkpoint_file.with_suffix(".tmp")
+    with open(tmp_file, "wb") as file:
+        pickle.dump(state, file, protocol=pickle.DEFAULT_PROTOCOL)
+    tmp_file.rename(checkpoint_file)
+
+
+class ProfetTask(Task):
+    def __init__(
+        self,
+        benchmark: str,
+        task_id: Optional[int]=0,
+        input_dir: Union[Path, str] = None,
+        rng: np.random.RandomState = None,
+        seed: int = 1,
+        checkpoint_dir: Union[Path, str] = None,
+        max_trials: int = 100,
+        train_config: MetaModelTrainingConfig = None,
+        **kwargs,
+    ):
+        if rng is None:
+            rng = np.random.RandomState(seed)
+        task_id = task_id or 0
+        super().__init__(
+            task_id=task_id,
+            rng=rng,
+            seed=seed,
+            max_trials=max_trials,
+            fixed_dims=kwargs,
+        )
+        self.benchmark = benchmark
+        if not input_dir:
+            input_dir = Path(os.environ.get("DATA_DIR")) / "profet"
+        else:
+            input_dir = Path(input_dir)
+
+        if checkpoint_dir is None:
+            checkpoint_dir = input_dir / "checkpoints"
+        # TODO: Save the train config in the `configuration` of the task.
+        self.train_config = train_config or MetaModelTrainingConfig()
+        task_hash_params = dict(
+            benchmark=benchmark,
+            task_id=task_id,
+            rng=rng,
+            seed=seed,
+            **asdict(self.train_config),
+        )
+        task_hash = compute_identity(**task_hash_params)
+
+        filename = f"{task_hash}.pkl"
+
+        checkpoint_file = Path(checkpoint_dir) / filename
+        logger.debug(f"Checkpoint file for this task: {checkpoint_file}")
+
+        # The config for the training of the meta-model.
+        # TODO: This should probably be a part of the hash for the "id" of the task!
+        self.train_config = train_config
+
+        if os.path.exists(checkpoint_file):
+            self.net, self.h = load_task_network(checkpoint_file)
+        else:
+            logger.info(
+                f"Checkpoint file {checkpoint_file} doesn't exist: re-training the meta-model."
+            )
+            # TODO: Figure out where/how to set the logger config.
```

Comment:
If using python's logger then we can rely on global config.

Reply:
Is it alright to use a logger per module? Should I be using `logger.info`/`logger.debug` or `logging.info`/`logging.debug` ?